transport/http/constants.go

```diff
@@ -22,20 +22,24 @@ package http
 
 const (
 	// CallerHeader is the HTTP header used to indiate the service doing the calling
-	CallerHeader = "RPC-Caller"
+	CallerHeader = "Rpc-Caller"
```

Comment:
We're case insensitive, but HTTP headers are normalized to "Title-Case" by Go so the default values we use should be Title-Cased already.